app/javascript/hyacinth_ui_v1/components/projects/permissions/PermissionsEditor.js

```diff
@@ -0,0 +1,255 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useQuery, useMutation } from '@apollo/react-hooks';
+import { Button, Form, Table } from 'react-bootstrap';
+import produce from 'immer';
+import Select from 'react-select';
+import { remove as arrRemove } from 'lodash/array';
+
+import GraphQLErrors from '../../ui/GraphQLErrors';
+import { getProjectPermissionActionsQuery, getProjectPermissionsQuery, updateProjectPermissionsMutation } from '../../../graphql/projects';
+import { getUsersQuery } from '../../../graphql/users';
+import CancelButton from '../../layout/forms/CancelButton';
+
+function PermissionsEditor(props) {
+  const { readOnly } = props;
+
+  const { projectStringKey } = props;
+  const [permissionsChanged, setPermissionsChanged] = useState(false);
+  const [removedUserIds, setRemovedUserIds] = useState(new Set());
+  const [projectPermissions, setProjectPermissions] = useState([]);
+  const [currentAddUserSelection, setCurrentAddUserSelection] = useState(null);
+
+  const { loading: actionsLoading, error: actionsError, data: actionsData } = useQuery(
```

Comment:
This is a nicer way to do the multiple queries than I had used (before I just broke things out into another component). Reads well.

Reply:
Thanks!